hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStateStore.java

```diff
@@ -545,6 +547,83 @@ public void overwriteRegions(List<RegionInfo> regionInfos, int regionReplication
     LOG.debug("Overwritten regions: {} ", regionInfos);
   }
 
+  /**
+   * Update region replicas if necessary by adding new replica locations or removing unused region
+   * replicas
+   */
+  public void updateRegionReplicas(TableName tableName, int oldReplicaCount, int newReplicaCount)
+    throws IOException {
+    if (newReplicaCount < oldReplicaCount) {
+      removeRegionReplicas(tableName, oldReplicaCount, newReplicaCount);
+    } else if (newReplicaCount > oldReplicaCount) {
+      addRegionReplicas(tableName, oldReplicaCount, newReplicaCount);
+    }
+  }
```

Comment:
If old and new  replica counts are equal, we just pass through -- that seems right.